packages/maskbook/src/plugins/Ideamarket/define.tsx

```diff
@@ -0,0 +1,58 @@
+import { PluginConfig, PluginStage, PluginScope } from '../types'
+import { usePostInfoDetails } from '../../components/DataSource/usePostInfo'
+import { getAllListings } from './api'
+import type { getAllListingsData } from './types'
+import LogoButton from './UI/LogoButton'
+import { isArray } from 'lodash-es'
+import Logo from '../../extension/options-page/DashboardComponents/MaskbookLogo'
+
+let ideaTokens: getAllListingsData[]
+
+const fetchAll = async () => {
+    const result = await getAllListings()
+
+    ideaTokens = result.ideaTokens
+}
+
+fetchAll()
+
+export const IdeamarketPluginDefine: PluginConfig = {
+    pluginName: 'Ideamarket',
+    identifier: 'ideamarket.io',
+    stage: PluginStage.Development,
+    scope: PluginScope.Internal,
+
+    postInspector: function Component(): JSX.Element | null {
+        const user = usePostInfoDetails('postBy').toText()
```

Comment:
```suggestion
        const user = usePostInfoDetails('postBy').userId
```

Why do you need to parse this object by yourself?